### PR TITLE
Create new "sanity" github action (just a few tests)

### PR DESCRIPTION
**Pull request by patimen**
Also add AI guidelines for better agent behavior and fix one weird edge case.

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -5,56 +5,21 @@ on:
   push:
     branches:
       - main
+    paths:
```
Comment (DimaOrekhovPS) [inline .github/workflows/integration.yml:8]: Can we exclude .md files and such? I expect a lot of changes in .md files in the upcoming months

File: .junie/guidelines.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+```
+/client-libs        # Client script to interact with the chain
+/cosmovisor         # Cosmovisor binaries
+/decentralized-api  # Api node
+/dev_notes          # Chain developer knowledge base
+/docs               # Documentation on specific aspects of the chain
+/inference-chain    # Chain node
+/prepare-local      # Scripts and configs for running local chain
+/testermint         # Integration tests suite
```
Comment (DimaOrekhovPS) [inline .junie/guidelines.md:24]: /local-test-net # Contains scripts and other files for running a local test net consisting of 3 nodes

Something like that

File: .github/workflows/integration.yml (hunk 1 above)
Comment (patimen) [reply .github/workflows/integration.yml:8]: There is no reliable way to exclude .md files in already included paths. There were some pretty hacky workarounds, I might look into them later.

**Review by DimaOrekhovPS — APPROVED** [review]
